src/Serialization/Rings.jl

```diff
@@ -0,0 +1,111 @@
+################################################################################
+#  non simpleton base rings
+
+function load_internal(s::DeserializerState, t::Type{Nemo.NmodRing}, dict::Dict)
+    modulus = load_type_dispatch(s, UInt64, dict[:modulus])
+
+    return t(modulus)
```

Comment:
Might as well encode the type explicitly?
```suggestion
function load_internal(s::DeserializerState, ::Type{Nemo.NmodRing}, dict::Dict)
    modulus = load_type_dispatch(s, UInt64, dict[:modulus])
    return NmodRing(modulus)
```